src/lib.rs

```diff
@@ -573,6 +573,12 @@ impl SharedSecret {
 
         Ok(SharedSecret(inner))
     }
+
+    pub fn clear(&mut self) {
```

Comment:
Maybe we can inline this to `Drop`? An issue I worry is that `[0u8; 32]` is not a valid shared secret, so `clear` might be misused.

Reply:
Hmm if you prefer I can remove the clear function and put it directly in Drop

Reply:
Yes please do!

Reply:
Done